plugins/cost-insights/api-report.md

```diff
@@ -178,10 +178,10 @@ export type BarChartProps = {
   resources: ResourceData[];
   responsive?: boolean;
   displayAmount?: number;
-  options?: Partial<BarChartData>;
-  tooltip?: ContentRenderer<TooltipProps>;
-  onClick?: RechartsFunction;
-  onMouseMove?: RechartsFunction;
+  options?: Partial<BarChartOptions>;
+  tooltip?: ContentType<ValueType, NameType>;
```

Comment:
Yeah here i'm wondering if this shouldn't just be a narrow, precise type like `{ title: string, description: string }` or whatever. Then that's the feature set we promise, for now, and extend it in the future if we ever deem it necessary.

The reason I'm bringing this up is, I'm essentially wondering if we can _remove_ as much explicit recharts-type-fiddling as possible. Look at it this way: if the type had been that in the first place, then the recharts update would not have had to make a breaking change to the public API surface here.

Reply:
You're talking about the `tooltip?: ContentType<ValueType, NameType>`?

Since it's expecting a function that returns a `ReactNode` I've created a `TooltipRenderer` type like the following:

```ts
export type TooltipRenderer = ((props: TooltipProps<string, string>) => ReactNode)
```

We wouldn't depend on the `ContentType` type anymore, only `TooltipProps` which are the available props for the tooltip